### PR TITLE
Footer - front-end

### PR DESCRIPTION
**Pull request by helenb**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1348054103)

### Description of Changes Made
Styling for the footer, along with pattern library yaml

### How to Test
Best tested in the pattern library: http://localhost:8000/pattern-library/render-pattern/patterns/organisms/footer/footer.html

To test the logos you can upload some to the navigation settings and then view any page (examples provided under screenshots)

### Screenshots

<details>
  <summary>Expand to see more</summary>

Logos for footer:

inhabit-carbon-neutral-badge-connected-blue ori
2022-top25_best-mid-sized-companies_1 original 1
certified_b_corporation_b_corp_logo_2022_black_
eoa_members_rgb_png_eoa_members_rgb_svg_brandma

Screenshot from pattern library:

Screenshot 2024-02-07 at 08 38 03

Logos on a page:
Screenshot 2024-02-07 at 08 38 31

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [x] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [x] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [x] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    {% include "patterns/atoms/motif-heading/motif-heading.html" with heading=contact_heading heading_level=2 classes="motif-heading--two motif-heading--static footer-cta__heading" %}
+
+    <p class="footer-cta__text">{{ contact_text }} </p>
+
+    <div class="footer-cta__grid">
+        {% include "patterns/atoms/avatar/avatar.html" with classes="footer-cta__avatar avatar--smallest" %}
+
+        <p class="footer-cta__cite">
+            <span class="footer-cta__name">{{ contact_name }}</span>
+            <span class="footer-cta__role">{{ contact_role }}</span>
```
Comment (chris-lawton) [inline tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html:11]: In the pattern lib, the spacing looks quite tight between these 2:

image

Design:

image

File: tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        </p>
+
+        <div class="footer-cta__button-wrapper">
+            <a href="contact_link" class="button footer-cta__button">Get in touch</a>
```
Comment (chris-lawton) [inline tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html:15]: Should `contact_link ` be a variable?

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html [hunk 3]
```diff
@@ -1,53 +1,58 @@
...
-            <a aria-label="Follow us on Instagram" href="https://www.instagram.com/torchboxltd/">
-                <svg width="24" height="24" aria-hidden="true">
-                    <use xlink:href="#instagram" />
-                </svg>
-            </a>
-        </li>
-    </ul>
+            <ul class="footer__socials">
+                <li>
+                    <a aria-label="Follow us on Twitter" href="https://twitter.com/torchbox" class="footer__social-link">
```
Comment (chris-lawton) [inline tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:15]: These social anchors are missing our focus styles - they've got the browser default at the mo : )

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html [hunk 4]
```diff
@@ -1,53 +1,58 @@
...
+
+    <div class="grid footer__bottom">
+        <div class="grid__footer-logos">
+            {% with global_settings=settings.navigation.NavigationSettings %}
+                {% if global_settings.footer_logos %}
+                    <ul class="footer__logos">
+                        {% for item in global_settings.footer_logos %}
+                            {% with footer_logo=item.value.image %}
+                                <li>
+                                    {% image footer_logo max-100x100 format-webp loading="lazy" class="footer__logo" %}
```
Comment (chris-lawton) [inline tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:47]: There's an option in Wagtail to add a link to these - is this being removed or do we need to account for that in the FE?

image

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html (hunk 3 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:15]: There's a mixin for the focus styles here: https://github.com/torchbox/torchbox.com/blob/integration/sprint-3/tbx/static_src/sass/config/_mixins.scss#L122

**Review by chris-lawton — APPROVED** [review]
Thanks @helenb - looks great. Really nice work. Left a few comments for review but nothing huge.

File: tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html (hunk 1 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html:11]: ah nice spot - fixed in 85c83e42db2fc924f4b1d4f42a85884af819aa0d

File: tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html (hunk 2 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/footer-cta/footer-cta.html]: fixed in 88e320342b19f7164c5148a950419e8012b98129

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html (hunk 3 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:15]: fixed in bd61252d5f5344393bd91350cb5ca611d96828f7

File: tbx/project_styleguide/templates/patterns/organisms/footer/footer.html (hunk 4 above)
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/organisms/footer/footer.html:47]: That link option is being removed in the back-end as at the moment it is used inconsistently.

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html [hunk 5]
```diff
@@ -9,15 +9,15 @@
     {% endif %}
 
     {% if value.person %}
-        <cite class="contact-cta__cite">
+        <p class="contact-cta__cite">
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html:12]: @chris-lawton I forgot to say that I removed these `<cite>` attributes as they weren't being used correctly - see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/cite

**Pull request merged**